src/test/scala/spark/dicom/TestDicomFileFormat.scala

```diff
@@ -1,55 +1,99 @@
+
+trait WithSpark {
+  var spark = {
     val spark = SparkSession.builder.master("local").getOrCreate
-    spark.sparkContext.setLogLevel("ERROR")
-    try {
-      val df = spark.read
-        .format("dicom")
-        .load(
-          "src/test/resources/Pancreatic-CT-CBCT-SEG/Pancreas-CT-CB_001/07-06-2012-NA-PANCREAS-59677/1.000000-BSCBLLLRSDCB-27748/1-1.dcm"
-        )
-        .select("path", "length", "content")
-      df.show
-    } finally {
-      spark.stop
-    }
+    spark.sparkContext.setLogLevel(Level.ERROR.toString())
+    spark
+  }
+}
+
+class TestDicomFileFormat
+    extends AnyFlatSpec
+    with WithSpark
+    with BeforeAndAfterAll {
+
+  val logger = LogManager.getLogger("TestDicomFileFormat");
+  logger.setLevel(Level.DEBUG)
+
+  override protected def afterAll(): Unit = {
+    spark.stop
+  }
+
+  "Spark" should "read DICOM files" in {
+    val df = spark.read
+      .format("dicom")
+      .load(
+        "src/test/resources/Pancreatic-CT-CBCT-SEG/Pancreas-CT-CB_001/07-06-2012-NA-PANCREAS-59677/1.000000-BSCBLLLRSDCB-27748/1-1.dcm"
+      )
+      .select("path", "length", "content")
+    assert(df.first().getAs[Long]("length") == 2384336)
   }
 
   "Spark" should "stream DICOM files" in {
-    val spark = SparkSession.builder.master("local").getOrCreate
-    spark.sparkContext.setLogLevel("ERROR")
-    try {
-      val df = spark.readStream
-        .schema(DicomFileFormat.SCHEMA)
-        .format("dicom")
-        .load(
-          "src/test/resources/Pancreatic-CT-CBCT-SEG/Pancreas-CT-CB_001/*/*/*"
-        )
-
-      val queryName = "testStreamDicom"
-      val query = df.writeStream
-        .trigger(Trigger.Once)
-        .format("memory")
-        .queryName(queryName)
-        .start
-
-      query.processAllAvailable
-      spark.table(queryName).show
-
-    } finally {
-      spark.stop
-    }
+    val df = spark.readStream
+      .schema(DicomFileFormat.SCHEMA)
+      .format("dicom")
+      .load(
+        "src/test/resources/Pancreatic-CT-CBCT-SEG/Pancreas-CT-CB_001/07-06-2012-NA-PANCREAS-59677/201.000000-PANCREAS DI iDose 3-97846/*"
+      )
+
+    val queryName = "testStreamDicom"
+    val query = df.writeStream
+      .trigger(Trigger.Once)
+      .format("memory")
+      .queryName(queryName)
+      .start
+
+    query.processAllAvailable
+    val outDf = spark.table(queryName)
+    assert(outDf.count() == 79)
+  }
+
+  "Spark" should "write a DICOM file" in {
+    val df = spark.read
+      .format("dicom")
+      .load(
+        "src/test/resources/Pancreatic-CT-CBCT-SEG/Pancreas-CT-CB_001/07-06-2012-NA-PANCREAS-59677/1.000000-BSCBLLLRSDCB-27748/1-1.dcm"
+      )
+      .select("path", "length", "content")
+
+    val tmpDir = Files.createTempDirectory("some-dicom-files")
+    tmpDir.toFile.delete // need to delete since Spark handles creation
+    val tmpPath = new Path(tmpDir.toUri)
+    val outPath = new Path(tmpDir.resolve("1-1.dcm").toUri)
+
+    // write to single
+    df.repartition(1)
```

Comment:
Repartitioning along path column yields wanted effect of restoring file structure
Cannot ideally be added to API but fine for moment